src/OracleService/Protocols/OracleHttpsProtocol.cs

```diff
@@ -22,7 +22,10 @@ namespace Neo.Plugins
 {
     class OracleHttpsProtocol : IOracleProtocol
     {
-        private readonly HttpClient client = new(new HttpClientHandler() { AllowAutoRedirect = false });
+        private readonly HttpClient client = new(new HttpClientHandler() { AllowAutoRedirect = false })
+        {
+            MaxResponseContentBufferSize = OracleResponse.MaxResultSize
+        };
```

Comment:
We can't just revert this, the whole point of this PR is to restrict the amount of data that can be read from the response stream to avoid OOM.

Reply:
> We can't just revert this, the whole point of this PR is to restrict the amount of data that can be read from the response stream to avoid OOM.

OK. The problem here is that we can't achieve
```
            if (!message.Content.Headers.ContentLength.HasValue || message.Content.Headers.ContentLength > OracleResponse.MaxResultSize)
                return (OracleResponseCode.ResponseTooLarge, null);
``` 
because it will throw exception by `System` from .NET and it can't be seen since we con't `catch (Exception ex)`. Finally we will always get `OracleResponseCode.Timeout`. Any other good idea?
PS: We shouldn't return `OracleResponseCode.Timeout` as default when catch exception. It doesn't make sense, cofused me a lot. I think it should be `OracleResponseCode.rror`.

Reply:
> Oh, I missed that commit. Can we test the resulting code?

Yes, already tested it before. The result here is my test result. It works well.
https://github.com/neo-project/neo-modules/pull/800#discussion_r1193521163